frontend/src/components/Aggregate.tsx

```diff
@@ -18,6 +21,42 @@ const Aggregate = () => {
     (state) => state.user.fceAggregation.numSemesters
   );
 
+  // Auxiliary local state required to handle the case when the field is (temporarily) blank
+  const [numSemsField, setNumSemsField] = React.useState<number | undefined>(
+    undefined
+  );
+
+  // To update the global state
+  React.useEffect(() => {
+    setNumSemsField(numSemesters);
+  }, [numSemesters]);
+
+  /* Needed to prevent non-numeric input from being entered by the user. */
+  const handleNumSemsFieldKeyDown: React.KeyboardEventHandler<
+    HTMLInputElement
+  > = (e) => {
+    if (!numeric.test(e.key)) {
+      e.preventDefault();
+    }
+  };
+
+  const handleNumSemsFieldChange: React.ChangeEventHandler<HTMLInputElement> = (
+    e
+  ) => {
+    if (numSemsDispatchableRegex.test(e.target.value)) {
+      setNumSemesters(parseInt(e.target.value));
+    } else if (e.target.value === "") {
+      setNumSemsField(undefined);
+    }
+  };
+
+  // Restore the last valid value if the field is left blank and unfocused
+  const handleNumSemsFieldBlur: React.FocusEventHandler<
+    HTMLInputElement
+  > = () => {
+    if (numSemsField === undefined) setNumSemsField(numSemesters);
+  };
+
```

Comment:
I don't think there's a way to get around needing an auxiliary state, so this is fine. You might be able to streamline it though: the input has `numSemsField`. Once the input is unfocused, we either restore to the redux state if its value is nonnumeric, or update the redux state if its value is a valid number.

That way, we don't need to override the keyDown event. I tested it locally and currently, there are some odd behaviors where after backspacing, I can't type any numbers anymore. I think the above method would be more robust.